doc/developer/platform/formalism.md

```diff
@@ -13,57 +13,370 @@ It does not intend to describe the **current behavior** of the system.
-    The updates are not discarded, but their times may have been advanced to allow cancelation and consolidation of updates to occur.
-    This is analogous to "version vacuuming" in multiversioned databases.
-    As the system operates, the `since` frontier may advance to allow compaction of updates.
-    This happens subject to constraints, and only when correctness permits.
-
-    We think of `since` as the "read frontier": times greater or equal to it may yet be correctly read from the TVC.
-
-These two frontiers mean that for any `time` greater or equal to `since`, but not greater or equal to `upper`, we can exactly reconstruct the collection at `time`.
-If `time` is not greater or equal to `since`, we may have permanently lost historical distinctions that allow us to know the contents at `time`.
-If `time` is greater or equal to `upper`, we may still receive changes to the collection that occur at `time`.
-The two frontiers never move backward, and often move forward as the system operates.
-
-At each moment, Materialize's representation of a TVC is as a set of update triples `(data, time, diff)` and the two frontiers `since` and `upper`.
-We call this pair of set of updates and `(since, upper)` frontiers a *partial TVC* or pTVC.
-
-## Evolution
-
-Materialize maintains a (partial) map from `GlobalId`s to pTVCs.
-A `GlobalId` is never re-used to mean a different TVC.
+You may wish to consult the
+[VLDB](https://vldb.org/pvldb/vol13/p1793-mcsherry.pdf) and
+[CIDR](https://github.com/TimelyDataflow/differential-dataflow/blob/master/differentialdataflow.pdf)
+papers on differential dataflow, as well as the [DD
+library](https://github.com/TimelyDataflow/differential-dataflow).
+
+# In a nutshell
+
+A *time-varying collections* (TVC) is a map of partially ordered *times* to
+*collection versions*, where each collection version is a multiset of typed
+*pieces of data*. Because the past and future of a TVC may be unknown,
+Materialize represents a TVC as a *partial TVC* (pTVC), which is an accurate
+representation of a TVC during a range of times bounded by two *frontiers*.
+
+To avoid storing and processing the entire collection version for every single
+time, Materialize uses sets of *updates* as a differential representation of a
+pTVC. These updates are triples of `[data, time, diff]`: the piece of data
+being changed, the time of the change, and the change in multiplicity of that
+data. Materialize can *compact* a pTVC by merging updates from the past, and
+*append* new updates to a pTVC as they arrive. Compaction advances the lower
+*since frontier* of the pTVC, and appending advances the *upper frontier*. Both
+compact and append preserve correctness: these differential representations of
+a pTVC are faithful to the "real" TVC between `since` and `upper`.
+
+Materialize identifies each TVC by a unique `GlobalId`, and maintains a mutable map of `GlobalId`s to the current pTVC representing that `GlobalID`'s TVC.
```

Comment:
> A GlobalID identifies a single TVC for all time (is this true? I think your comment above might mean... no?)

I meant it to be a single TVC for all time. I didn't realize I implied otherwise, so I might be miscommunicating.

The next two point (single pTVC and no first-class representation) are both spot on!

In MZ there is a `GlobalId`. That id is used in a few ways. For example, the catalog will store things like
```
CREATE VIEW foo AS
SELECT * FROM bar
```
using ids for `foo` and for `bar`. These correspond, imo, to the TVCs. We don't have the TVC at hand, but we are naming them and their relationships.

When you ship a dataflow to COMPUTE, you use the ids (say for `foo` and `bar` as in the query) to name the things you want to relate. At any moment one can resolve a pTVC associated with each of them, but their meaning is imo actually about the TVCs. E.g. at any moment maybe the pTVC for `foo` isn't yet caught up to `bar`. But it should be, because what we actually have is a relationships between TVCs, more than specific pTVCs.

The TVC is never reified in memory, but it is a real concept that some parts of the system mean to speak about (rather than speak about the associated pTVCs).

Reply:
Thank you! This all makes sense. I'm going to add a section to the formalism that talks about GlobalIds specifically, and brings in some of these concepts, just in case they're not clear in the rest of the formalism.